lib/compat/wordpress-6.3/class-gutenberg-rest-global-styles-controller-6-3.php

```diff
@@ -0,0 +1,184 @@
+						'author'    => array(
+							'display_name' => get_the_author_meta( 'display_name', $post->post_author ),
+							'avatar_url'   => get_avatar_url(
+								$post->post_author,
+								array(
+									'size' => 24,
+								)
+							),
+						),
+					);
+				}
+			}
+		}
+		return rest_ensure_response( $revisions );
+	}
+
+	/**
+	 * Prepare a global styles config output for response.
+	 *
+	 * @since 5.9.0
+	 * @since 6.2 Handling of style.css was added to WP_Theme_JSON.
+	 * @since 6.3 Adds version-history to the response object.
+	 *
+	 * @param WP_Post         $post Global Styles post object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$raw_config                       = json_decode( $post->post_content, true );
+		$is_global_styles_user_theme_json = isset( $raw_config['isGlobalStylesUserThemeJSON'] ) && true === $raw_config['isGlobalStylesUserThemeJSON'];
+		$config                           = array();
+		if ( $is_global_styles_user_theme_json ) {
+			$config = ( new WP_Theme_JSON_Gutenberg( $raw_config, 'custom' ) )->get_raw_data();
+		}
+
+		// Base fields for every post.
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( rest_is_field_included( 'id', $fields ) ) {
+			$data['id'] = $post->ID;
+		}
+
+		if ( rest_is_field_included( 'title', $fields ) ) {
+			$data['title'] = array();
+		}
+		if ( rest_is_field_included( 'title.raw', $fields ) ) {
+			$data['title']['raw'] = $post->post_title;
+		}
+		if ( rest_is_field_included( 'title.rendered', $fields ) ) {
+			add_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+
+			$data['title']['rendered'] = get_the_title( $post->ID );
+
+			remove_filter( 'protected_title_format', array( $this, 'protected_title_format' ) );
+		}
+
+		if ( rest_is_field_included( 'settings', $fields ) ) {
+			$data['settings'] = ! empty( $config['settings'] ) && $is_global_styles_user_theme_json ? $config['settings'] : new stdClass();
+		}
+
+		if ( rest_is_field_included( 'styles', $fields ) ) {
+			$data['styles'] = ! empty( $config['styles'] ) && $is_global_styles_user_theme_json ? $config['styles'] : new stdClass();
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		if ( rest_is_field_included( '_links', $fields ) || rest_is_field_included( '_embedded', $fields ) ) {
+			$links = $this->prepare_links( $post->ID );
+			if ( $is_global_styles_user_theme_json ) {
+				$revisions                = wp_get_latest_revision_id_and_total_count( $post->ID );
```

Comment:
Do revisions have revisions?

Reply:
No, I'm creating a new endpoint to grab revisions for global styles.

What I wanted to do here was update the response for the [existing route](https://github.com/ramonjd/wordpress-develop/blob/b24f990f4f7789cee65628ef090a9e68a2798b5b/src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php#L87-L87) to include the URL to the new revisions endpoint in  `version-history` links.

I was trying to do it according to the pattern in the [Posts controller](https://github.com/ramonjd/wordpress-develop/blob/b24f990f4f7789cee65628ef090a9e68a2798b5b/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2068-L2068)

I see now that I could have rather overwritten [prepare_links](https://github.com/ramonjd/wordpress-develop/blob/b24f990f4f7789cee65628ef090a9e68a2798b5b/src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php#L429-L429), which is a lot less code. I'll do that for now.